### PR TITLE
Makes "manual" the default build strategy

### PR DESCRIPTION
**Pull request by conorsch**
By default the build role will now stop after preparing the environment, prompting the admin to continue manually with the following message:

> Build environment ready. Kernel source is available in {{ linux_source_directory }} . Run `make menuconfig` in that directory to configure the build, then `make-kpkg kernel_image` or similar to begin compiling the kernel.

The manual strategy can be overridden via `grsecurity_build_strategy=digitalocean`, with support for more providers to come, as described in #19. 
### Major changes
- defaults to `manual` build strategy
- new default var `include_ubuntu_overlay` set to `false`
- separates the `compile.yml` task list into:
  - `prepare_source_directory.yml` (with new `prepare` tag for easy targeting)
  - `configure.yml` (with new `configure` tag for easy targeting)
  - `compile.yml` (with new `compile` tag for easy targeting)
- includes the build strategy string in the filename for the deb package

### CODE REVIEW

**Comment by conorsch** [comment]
@ageis Can I get an r in the :house:?

Once these changes are merged, I'll add dynamic URL fetching via a separate PR, and after that we can take a crack at folding in multiple config files for different platforms.

**Comment by ageis** [comment]
Hmm interesting approach - manualness is never ideal but this is probably just what we have to do. We can discuss relative merits and drawbacks later, I kinda wanna diff some more configs and do more investigation. This reads well but let's wait til I can fully test it before merge - will do this afternoon.

On a related note I revisited the other providers yesterday, tried doing everything from scratch and still came up with kernel .debs too small. Still working on that.

**Comment by conorsch** [comment]
The default-to-manual approach is the sanest default I can come up with: it simply won't break things. For automagic mode, simply set `grsecurity_build_strategy=true` while running the playbook and you're all set.

We'll handle how best to ship multiple configs in a separate PR (or discuss in #19). Holler if changes necessary here.

File: ansible/roles/build-grsec-kernel/tasks/compile.yml [hunk 1]
```diff
@@ -1,37 +1,10 @@
...
-- name: Generate config, accepting default choices.
-  shell: yes '' | make-kpkg configure
-  args:
-    chdir: "{{ linux_source_directory }}"
-
 - name: Build the grsecurity-patched kernel.
-  command: fakeroot make-kpkg --initrd kernel_image
+  command: >
+    fakeroot make-kpkg
+    --revision 10.00.{{ grsecurity_build_strategy }}
```
Comment (ageis) [inline ansible/roles/build-grsec-kernel/tasks/compile.yml:5]: Just curious, what's your thinking behind putting in the revision?

**Comment by ageis** [comment]
Tested this (incl. the PR https://github.com/conorsch/grsec/pull/2 I submitted to your branch that updates stable patches and Linux to 3.14.54) and it works for me.

File: ansible/roles/build-grsec-kernel/tasks/compile.yml (hunk 1 above)
Comment (conorsch) [inline ansible/roles/build-grsec-kernel/tasks/compile.yml:5]: When building for multiple platforms, the `fetch` task would clobber previously built debs due to the filename being identical. Inserting the build strategy into the filename makes it clear which packages should be installed on which hosts.

**Pull request merged**